paddle/fluid/framework/data_type.h

```diff
@@ -22,46 +22,59 @@ limitations under the License. */
 namespace paddle {
 namespace framework {
 
+template <typename T>
+struct DataTypeTrait {};
+
+// Stub handle for void
```

Comment:
Add DataTypeTrait for void because we allow the user invoke `tensor.data<void>()` and `tensor.mutable_data<void>`, we will never use `sizeof(void)` because it will be a shortcut like `std::is_same<void, T> || SizeOf(DataTypeTrait<void>::DataType)`. However, we must write this template specification here.

Reply:
It's surprise that gcc define the sizeof(void). Although it is not defined in cpp standards.
```cpp
echo 'int main() { printf("%d", sizeof(void)); }' | g++ -xc -w - && ./a.out
```
it gives 1.